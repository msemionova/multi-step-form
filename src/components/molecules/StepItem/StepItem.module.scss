@import '../../../styles/main';

.step {
  display: flex;
  align-items: center;
  gap: 1.4rem;

  h2, span {
    text-transform: uppercase;
    line-height: 1;
  }

  h2 {
    font-size: 1.4rem;
    color: var(--color-gray-background);
    letter-spacing: .1rem;
  }

  span {
    font-size: 1.2rem;
    color: var(--color-gray-line);
  }

  &:not(:last-child) {
    margin-bottom: 2.5rem;
  }
}
