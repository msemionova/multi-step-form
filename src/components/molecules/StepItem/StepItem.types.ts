export type StepItemProps = {
  number: number,
  title: string,
  isActive?: boolean
}
