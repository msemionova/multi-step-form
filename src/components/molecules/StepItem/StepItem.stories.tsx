import type { Meta, StoryObj } from '@storybook/react';

import StepItem from './StepItem';

const meta: Meta<typeof StepItem> = {
  component: StepItem,
};

export default meta;
type Story = StoryObj<typeof StepItem>;

export const Default: Story = {
  args: {
    text: '1'
  }
};
