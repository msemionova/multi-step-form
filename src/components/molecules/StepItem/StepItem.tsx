import styles from './StepItem.module.scss';
import { StepItemProps } from './StepItem.types';

import CircleNumber from '../../atoms/CircleNumber/CircleNumber';

export default function StepItem({ number, title, isActive = false }: StepItemProps) {
  const label = `Step ${number}`;

  return (
    <div className={styles.step}>
      <CircleNumber isActive={isActive}>{number}</CircleNumber>
      <div>
        <span>{label}</span>
        <h2>{title}</h2>
      </div>
    </div>
  );
}
