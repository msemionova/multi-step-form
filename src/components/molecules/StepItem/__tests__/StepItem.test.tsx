import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import StepItem from '../StepItem';

describe('<StepItem />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<StepItem number={1} title='Test'></StepItem>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
