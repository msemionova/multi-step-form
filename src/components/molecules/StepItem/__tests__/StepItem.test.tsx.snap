// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepItem /> renders correctly 1`] = `
<div
  className="step"
>
  1
</div>
`;
